apis/v1alpha2/gatewayclass_types.go

```diff
@@ -0,0 +1,205 @@
+	// The format of this field is DOMAIN "/" PATH, where DOMAIN is a valid
+	// Kubernetes name (https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names)
+	// and PATH is a valid HTTP path as defined by RFC 3986.
+	//
+	// Support: Core
+	//
+	// +kubebuilder:validation:MaxLength=253
+	Controller string `json:"controller"`
+
+	// ParametersRef is a reference to a resource that contains the configuration
+	// parameters corresponding to the GatewayClass. This is optional if the
+	// controller does not require any additional configuration.
+	//
+	// ParametersRef can reference a standard Kubernetes resource, i.e. ConfigMap,
+	// or an implementation-specific custom resource. The resource can be
+	// cluster-scoped or namespace-scoped.
+	//
+	// If the referent cannot be found, the GatewayClass's "InvalidParameters"
+	// status condition will be true.
+	//
+	// Support: Custom
+	//
+	// +optional
+	ParametersRef *ParametersReference `json:"parametersRef,omitempty"`
+
+	// Description helps describe a GatewayClass with more details.
+	//
+	//
+	// +kubebuilder:validation:MaxLength=64
+	// +optional
+	Description *string `json:"description,omitempty"`
+}
+
+// ParametersReference identifies an API object containing controller-specific
+// configuration resource within the cluster.
+type ParametersReference struct {
+	// Group is the group of the referent.
+	//
+	// +kubebuilder:validation:MaxLength=253
+	Group string `json:"group"`
+
+	// Kind is kind of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Kind string `json:"kind"`
+
+	// Name is the name of the referent.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+
+	// Scope represents if the referent is a Cluster or Namespace scoped resource.
+	// This may be set to "Cluster" or "Namespace".
+	// +kubebuilder:validation:Enum=Cluster;Namespace
+	// +kubebuilder:default=Cluster
+	// +optional
+	Scope *string `json:"scope,omitempty"`
+
+	// Namespace is the namespace of the referent.
+	// This field is required when scope is set to "Namespace" and ignored when
+	// scope is set to "Cluster".
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	// +optional
+	Namespace *string `json:"namespace,omitempty"`
+}
+
+// GatewayClassConditionType is the type for status conditions on
+// Gateway resources. This type should be used with the
+// GatewayClassStatus.Conditions field.
+type GatewayClassConditionType string
+
+// GatewayClassConditionReason defines the set of reasons that explain why a
+// particular GatewayClass condition type has been raised.
+type GatewayClassConditionReason string
+
+const (
+	// This condition indicates whether the GatewayClass has been admitted by
+	// the controller requested in the `spec.controller` field.
+	//
+	// This condition defaults to False, and MUST be set by a controller when it
```

Comment:
I think `False` is appropriate as a default here. It can not possibly have been admitted (at least not in a conformant way) unless the controller has actually flipped this to `True`.

Reply:
I think I agree with Dan.  True is obviously meaningful, but you specifically say below that the controller should set it to false, which means false has meaning.  Defaulting it to false dilutes that.

Reply:
Will be fixed by #839